### PR TITLE
periphery 2.21.2 (new formula)

### PR DESCRIPTION
**Pull request by ileitch**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by chenrui333** [comment]
@ileitch since periphery needs xcodeproject, maybe you should drop the linux support for the formula?

**Comment by ileitch** [comment]
> @ileitch since periphery needs xcodeproject, maybe you should drop the linux support for the formula?

It also works on Linux with just SPM, but I've had to disable the formula test on Linux because the test image appears to be missing `libIndexStore.so`. Hopefully that's acceptable 🤞.

File: Formula/p/periphery.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  url "https://github.com/peripheryapp/periphery/archive/refs/tags/2.21.2.tar.gz"
+  sha256 "97e399df17d1e681703c5c8852e50b40c056a98b41dc4c615011c048b8348dbb"
+  license "MIT"
+  head "https://github.com/peripheryapp/periphery.git", branch: "master"
+
+  livecheck do
+    url :stable
+    strategy :github_latest
+  end
+
```
Comment (chenrui333) [inline Formula/p/periphery.rb:13]: the default livecheck should be good enough
```suggestion
```

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
> > @ileitch since periphery needs xcodeproject, maybe you should drop the linux support for the formula?
> 
> It also works on Linux with just SPM, but I've had to disable the formula test on Linux because the test image appears to be missing `libIndexStore.so`. Hopefully that's acceptable 🤞.

Sounds good 👍

File: Formula/p/periphery.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  depends_on xcode: ["15.4", :build]
+  depends_on macos: :sonoma
```
Comment (chenrui333) [inline Formula/p/periphery.rb:15]: would swift 6 work for ventura?

Comment (ileitch) [reply Formula/p/periphery.rb]: I bumped it to `:ventura`  as that's the minimum version supported by the [Package.swift](https://github.com/peripheryapp/periphery/blob/master/Package.swift#L177).

Comment (chenrui333) [reply Formula/p/periphery.rb]: bumping to `:ventura` actually wont work

```
==> SKIPPED periphery
Warning: A full installation of Xcode.app 15.4 is required to compile
this software. Installing just the Command Line Tools is not sufficient.

Xcode 15.4 cannot be installed on macOS 13. 
```

https://github.com/Homebrew/homebrew-core/actions/runs/12508037405/job/34895612061#step:3:40

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
Looks good now, merging!

**Comment by chenrui333** [comment]
@ileitch, thanks for your first contribution to Homebrew-core! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request merged**